### PR TITLE
gh-87526: Remove dead initialization from parse_abbr function

### PR DESCRIPTION
**Pull request by alexhenrie**
Straightforward dead code elimination.

https://bugs.python.org/issue43360

* Issue: gh-87526

### CODE REVIEW

**Review by furkanonder — APPROVED** [review]

**Comment by furkanonder** [comment]
LGTM.  @alexhenrie Can you add the news entry?

**Comment by arhadthedev** [comment]
> Can you add the news entry?

This change is invisible to users and irrelevant for contributors so adding https://github.com/python/cpython/labels/skip%20news.

File: Modules/_zoneinfo.c [hunk 1]
```diff
@@ -1709,7 +1709,7 @@ static Py_ssize_t
 parse_abbr(const char *const p, PyObject **abbr)
 {
     const char *ptr = p;
-    char buff = *ptr;
+    char buff;
```
Comment (vstinner) [inline Modules/_zoneinfo.c]: I suggest to move the variable declaration closer to where it's used: inside the if block at least.

Comment (alexhenrie) [reply Modules/_zoneinfo.c]: Done

**Comment by vstinner** [comment]
Merged, thanks.

**Pull request merged**